### PR TITLE
Changes US01 and US02 into the restructured file

### PR DESCRIPTION
**Pull request by rsukitha**
#4 US01 Dates before current date
#3 US02 Birth before marriage

### CODE REVIEW

File: GedcomParser.py [hunk 1]
```diff
@@ -104,13 +107,72 @@ def parse_valid_results(results):
...
+            validate_dates(date, fam)
             fam.divorced = date[3]
     return families, individuals
 
+def get_date(date):
+      """
+      Method to get the date from input and convert it to Date format
+      :param date -- Input line tuple in format (level, tag, valid, arg)      
+      """      
+      month = {"JAN": 1, "FEB":2, "MAR":3, "APR":4, "MAY":5, "JUN":6, "JUL":7, "AUG":8, "SEP":9, "OCT":10, "NOV":11, "DEC":12}
```
Comment (CCorrado) [inline GedcomParser.py]: so this isn't needed...

https://docs.python.org/2/library/datetime.html#strftime-strptime-behavior

File: GedcomParser.py [hunk 2]
```diff
@@ -104,13 +107,72 @@ def parse_valid_results(results):
...
+      """
+      Method to validate if the birth, death, marriage, divorce dates are before today
+      :param date -- Input line tuple in format (level, tag, valid, arg) 
+      :param current_indi_id -- Individual object
+      """   
+      current_date = datetime.datetime.now()
+      year = current_date.year
+      month = current_date.month
+      day = current_date.day
+      today = datetime.date(year, month, day)
```
Comment (CCorrado) [inline GedcomParser.py]: you just need this:
```
today = datetime.datetime.now()
```

File: GedcomParser.py [hunk 3]
```diff
@@ -104,13 +107,72 @@ def parse_valid_results(results):
...
+      
+
+      given_date = get_date(date)
+
+      if given_date > today:
+            print("Invalid! Birth, Marriage, Divorce or Death dates should not be after today. User ID is: ", current_indi_id.id, " and today is: ", today)            
+      else:
+            pass
+
+def validate_marriage_date(birth, marriage, current_fam_id):
```
Comment (CCorrado) [inline GedcomParser.py]: these methods probably belong inside of the Family class? so you can do.. 
```
def validate_marriage_date(self, birth):
    today = datetime.datetime.now()
    return (self.marriage_date > today or self.marriage_date < birth):
```

**Comment by CCorrado** [comment]
fixes #3 and #4

**Comment by CCorrado** [comment]
@rsukitha do you want to implement the suggested changes?

**Comment by rsukitha** [comment]
Hello Chris,

Yeah I will be doing the suggested changes this week. Was quite stuck right
with my other subject assignments and tests last week.
Will be working on this code the next 2 days coming(Sunday and Monday)

Thanks,
Sukitha

**Comment by rsukitha** [comment]
Hello Chris,

I've updated the code! Please can you review it

**Pull request merged**